lib/Agrammon/DB/User.pm6

```diff
@@ -137,15 +137,16 @@ class Agrammon::DB::User does Agrammon::DB {
         return self;
     }
 
-    method exists {
+    method exists() {
+        my $uid;
         self.with-db: -> $db {
-            my $uid = $db.query(q:to/USER/, $!username).value;
+            $uid = $db.query(q:to/USER/, $!username).value;
                 SELECT pers_id AS id
                   FROM pers
                  WHERE pers_email = $1
             USER
-            return $uid;
         }
+        return $uid;
```

Comment:
I'm not sure if you need the variable here at all; doesn't `with-db` evaluate to whatever value as produced by the inner block?

Reply:
Either I am doing something wrong, or this is needed. This code
```
    method exists() {
        my $uid = self.with-db: -> $db {
            $db.query(q:to/USER/, $!username).value;
                SELECT pers_id AS id
                  FROM pers
                 WHERE pers_email = $1
            USER
        }
        note "uid=$uid";
        return $uid;
    }
```
outputs
```console
uid=The 'ForeignCode' class is a Rakudo-specific
implementation detail and has no serviceable parts inside
```

Reply:
Eww, that wants looking at, but not as part of this PR